src/kernel.cpp

```diff
@@ -75,19 +75,20 @@ bool IsFixedModifierInterval(unsigned int nTimeBlock)
 }
 
 // Get time weight
-int64_t GetWeight(int64_t nIntervalBeginning, int64_t nIntervalEnd)
+::int64_t GetWeight(int64_t nIntervalBeginning, ::int64_t nIntervalEnd)
 {
     // Kernel hash weight starts from 0 at the 30-day min age
     // this change increases active coins participating the hash and helps
     // to secure the network when proof-of-stake difficulty is low
     //
     // Maximum TimeWeight is 90 days.
 
-    return min(nIntervalEnd - nIntervalBeginning - nStakeMinAge, (int64_t)nStakeMaxAge);
+  //return min(nIntervalEnd - nIntervalBeginning - nStakeMinAge, (int64_t)nStakeMaxAge);
+    return min(nIntervalEnd - nIntervalBeginning, (int64_t)(nStakeMaxAge - nStakeMinAge) );
```

Comment:
=> I'm still not clear why you change coin time weight logic, with this change, maximum time weight is just 60 days, not 90 days anymore. I think we need to keep the POS logic same as 0.4.9